<!-- KITcar launch script for preprocessing as a node -->
<launch>
  <arg name="use_nodelet" default="false" />
  <arg name="car_name" default="$(env CAR_NAME)" />
  <arg name="start_activated" default="false" />
  <arg name="debug_mode" default="false" />
  <arg name="input_queue_size" default=" " />
  <arg name="output_queue_size" default=" " />
  <arg name="start_in_startbox_mode" default=" " />
  <arg name="camera_parameters_prefix" default=" " />

  <!-- if start as a node -->
  <arg unless="$(arg use_nodelet)" name="pkg" value="perception" />
  <arg unless="$(arg use_nodelet)" name="type" value="lane_detection" />
  <arg unless="$(arg use_nodelet)" name="args" value=" " />

  <!-- if start as a nodelet -->
  <arg if="$(arg use_nodelet)" name="nodelet_manager" />
  <arg if="$(arg use_nodelet)" name="pkg" value="nodelet" />
  <arg if="$(arg use_nodelet)" name="type" value="nodelet" />
  <arg if="$(arg use_nodelet)" name="args" value="load perception/lane_detection $(arg nodelet_manager)" />

  <env if="$(arg debug_mode)" name="ROSCONSOLE_CONFIG_FILE" value="$(find perception)/rosconsole/perception.conf"/>

  <include file="$(find car_specs)/launch/load_camera_parameters.launch" >
    <arg name="car_name" value="$(arg car_name)" />
    <arg if="$(eval arg('camera_parameters_prefix') != ' ')" name="prefix" value="$(arg camera_parameters_prefix)" />
  </include>

  <group ns="perception"> 
    <!-- loading package parameter -->
    <rosparam file="$(find perception)/param/$(arg car_name)/ego_vehicle.yaml" command="load" ns="default/ego_vehicle" />

    <node name="lane_detection" pkg="$(arg pkg)" type="$(arg type)" args="$(arg args)" output="screen">
      <remap from="lane_detection/preprocessed_image" to="preprocessing/preprocessed_image"/>
      <remap from="lane_detection/region_of_interest" to="preprocessing/region_of_interest"/>
      <remap from="lane_detection/auto_reset" to="/controller_interface/controller_interface/auto_reset" />

      <rosparam file="$(find perception)/param/lane_detection/default.yaml" command="load" ns="default" />
      <rosparam file="$(find perception)/param/lane_detection/free_ride_mode.yaml" command="load" ns="free_ride_mode" />
      <rosparam file="$(find perception)/param/lane_detection/obstacle_mode.yaml" command="load" ns="obstacle_mode" />
      <rosparam file="$(find perception)/param/lane_detection/parking_mode.yaml" command="load" ns="parking_mode" />
      <param name="start_activated" value="$(arg start_activated)" />
      <param name="debug_mode" value="$(arg debug_mode)" />
      <param if="$(eval arg('input_queue_size') != ' ')" name="default/input_queue_size" value="$(arg input_queue_size)" />
      <param if="$(eval arg('output_queue_size') != ' ')" name="default/output_queue_size" value="$(arg output_queue_size)" />
      <param if="$(eval arg('start_in_startbox_mode') != ' ')" name="default/line_tracking/start_box_handling/starts_with" value="$(arg start_in_startbox_mode)" />
    </node>
  </group>
</launch>
